benchmark_v2/sort_cases_benchmark.py

```diff
@@ -0,0 +1,179 @@
+    for ib in range(1, lgNv):
+        ii_bit = ak.uniform(N) > ab
+        jj_bit = ak.uniform(N) > (c_norm * ii_bit + a_norm * (~ii_bit))
+        ii = ii + ((2 ** (ib - 1)) * ii_bit)
+        jj = jj + ((2 ** (ib - 1)) * jj_bit)
+
+    yield "RMAT int64", (ii, jj)
+
+
+def block_sorted(N):
+    """
+    The concatenation of two sorted arrays of unequal length
+    The interleaving of two sorted arrays of unequal length
+
+    Most often occurs in array setops, where two arrays are
+    uniqued (via sorting), then concatenated and sorted
+    """
+    splitpoint = 0.4
+    Na = int(splitpoint * N)
+    Nb = N - Na
+    # Construct a and b such that:
+    #   1) Values overlap
+    #   2) a and b are sorted
+    a = ak.arange(Na)
+    b = ak.arange(Nb)
+    c = ak.concatenate((a, b), ordered=True)
+    yield "block-sorted concat int64", c
+
+    ci = ak.concatenate((a, b), ordered=False)
+    yield "block-sorted interleaved int64", ci
+
+
+def refinement(N):
+    """
+    Coargsort of two arrays, where the first is already sorted
+    but has many repeated values
+    """
+    groupsize = 100
+    a = ak.arange(N // 2) // groupsize
+    factor = 2**32 // a.max()
+    a *= factor
+    b = ak.randint(0, 2**32, N // 2)
+    yield "refinement int64", (a, b)
+
+
+def time_like(N):
+    """
+    Data like a datetime64[ns]:
+    - spanning 1 year
+    - with second granularity
+    - but stored with nanosecond precision
+    """
+    # seconds in a year
+    year_sec = 365 * 24 * 60 * 60
+    # offset to almost 2020 (yeah yeah, leap days)
+    twentytwenty = 50 * year_sec
+    # second-resolution timestamps spanning approx 2020-2021
+    a = ak.randint(0, year_sec, N) + twentytwenty
+    # stored as datetime64[ns]
+    a *= 10**9
+    yield "datetime64[ns]", a
+
+
+def IP_like(N):
+    """
+    Data like a 90/10 mix of IPv4 and IPv6 addresses
+    """
+    multiplicity = 10
+    nunique = N // (2 * multiplicity) if N >= (2 * multiplicity) else 1
+    # First generate unique addresses, then sample with replacement
+    u1 = ak.zeros(nunique, dtype=ak.int64)
+    u2 = ak.zeros(nunique, dtype=ak.int64)
+    v4 = ak.uniform(nunique) < 0.9
+    n4 = v4.sum()
+    v6 = ~v4
+    n6 = v4.size - n4
+    u1[v4] = ak.randint(0, 2**32, n4)
+    u1[v6] = ak.randint(-(2**63), 2**63, n6)
+    u2[v6] = ak.randint(-(2**63), 2**63, n6)
+    sample = ak.randint(0, nunique, N // 2)
+    IP1 = u1[sample]
+    IP2 = u2[sample]
+    yield "IP-like 2*int64", (IP1, IP2)
+
+
+GENERATORS = (random_uniform, power_law, rmat, block_sorted, refinement, time_like, IP_like)
+
+
+@pytest.mark.benchmark(group="AK_Sort_Cases")
+@pytest.mark.parametrize("gen", GENERATORS)
+@pytest.mark.parametrize("algo", SortingAlgorithm)
+def bench_sort_cases(benchmark, gen, algo):
+    """
+    Measure both performance of various sort cases
+    """
+    res = gen(pytest.prob_size)
+    data = res.send(None)[1]
+    benchmark.pedantic(do_argsort, args=(data, algo), rounds=pytest.trials)

```

Comment:
This won't work since these are generators, they could have more than one thing yielded (like in `block_sorted`, `power_law`, `random_uniform`)

At first I thought you were gonna have to do something like this:
```python
for _, data in gen(pytest.prob_size):
  benchmark.pedantic(do_argsort, args=(data, algo), rounds=pytest.trials)
```
But im realizing now that the parameterization would prob get messed up. Because it would record them all under the same generator (it doesn't know that `"block-sorted concat"` is different from `"block-sorted interleaved"`, it just knows they were both done using the parameter `block_sorted` so it will average their times)

I think the best way to do this in the new framework is to have a separate benchmark for each generator and then parameterize based on that, so the end result is more similar to the [graphs we have today](https://chapel-lang.org/perf/arkouda/16-node-cs-hdr/?startdate=2023/02/14&enddate=2023/03/21&configs=release,nightly&suite=sortcases). It's gonna be a bit tough because we want to use the same data in these cases (like in `power_law`, we don't wanna change the randomly generated data between the `"power-law float64"` and `"power-law int64"` cases. We want the only variable changed to be the type)